### ISSUE TITLE
[BUG]: when I input a not exist path, if it's number, 404 page will show index/index page

### ISSUE THREAD

**Issue by astinx** [phalcon/cphalcon#14926]
__Phalcon version__
> 4.05

__PHP version__
> 7.4.4

__this is my di container__
```php
$di->setShared('dispatcher', function () {
	//Create an EventsManager
	$eventsManager = new EventsManager();
	//Attach a listener
	$eventsManager->attach("dispatch:beforeException", function ($event, $dispatcher, $exception) {
		//Handle 404 not found exceptions
		if ($exception instanceof DispatchException) {
			$dispatcher->forward([
				'controller' => 'err',
				'action'     => 'notFound',
				'params'     => ['message' => $exception->getMessage()],
			]);
			return FALSE;
		}
	});

	$dispatcher = new MvcDispatcher();
	$dispatcher->setDefaultNamespace('\SRVX\Home\Controllers');
	//Bind the EventsManager to the dispatcher
	$dispatcher->setEventsManager($eventsManager);
	return $dispatcher;
});
```

when I enter a not exist url start with alphabet character, it will show 404 not found page like:
> xxx.com/abcd   

but I enter a character start with number 0~9, it will show index/index page
> xxx.com/1234

**Comment by wjiec** [comment]
same too, I found here

https://github.com/phalcon/cphalcon/blob/d3e94630156a0c4441bf5f72e6ed071941c5908b/phalcon/Mvc/Router.zep#L806

it checks numeric and fallback to default controller name by index. the same to action part.

what's a good idea?

**Comment by zsilbi** [comment]
Fixed in #15178
